host_core/lib/host_core/actors/actor_supervisor.ex

```diff
@@ -92,6 +92,59 @@ defmodule HostCore.Actors.ActorSupervisor do
     Map.get(all_actors(), public_key, [])
   end
 
+  @doc """
+  Ensures that the actor count is equal to the desired count by terminating instances
+  or starting instances on the host.
+  """
+  def scale_actor(public_key, desired_count, oci \\ "") do
+    current_instances = find_actor(public_key)
+    current_count = current_instances |> Enum.count()
+
+    # Attempt to retrieve OCI reference from running actor if not supplied
+    ociref =
+      cond do
+        oci != "" ->
+          oci
+
+        current_count >= 1 ->
+          ActorModule.ociref(current_instances |> List.first())
+
+        true ->
+          ""
+      end
+
+    diff = current_count - desired_count
+
+    cond do
+      # Current count is desired actor count
+      diff == 0 ->
+        {:ok}
```

Comment:
idiomatic here would be for the type of this fucntion to be `:ok | {:error, term}` rather than `{:ok} | {...}`  It's fine for the ok not to be in a tuple if there's no return value.